src/core/processing/qgsprocessingutils.cpp

```diff
@@ -155,26 +155,30 @@ class ProjectionSettingRestorer
 
 QgsMapLayer *QgsProcessingUtils::loadMapLayerFromString( const QString &string )
 {
-  if ( QFileInfo::exists( string ) )
-  {
-    // TODO - remove when there is a cleaner way to block the unknown projection dialog!
-    ProjectionSettingRestorer restorer;
-    ( void )restorer; // no warnings
+  QStringList components = string.split( '|' );
```

Comment:
Can you pull this change out to a separate PR? I have some questions/concerns about it, but the concatenator can be merged as is.